apps/docs/stories/architecture-decision-records/0007-why-do-we-transpile-web-components-for-react-and-vue.stories.mdx

```diff
@@ -0,0 +1,40 @@
+import { Meta } from '@storybook/addon-docs'
+
+<Meta title='Docs/Architecture Decision Records/ADR 0007: Why do we transpile Web Components for React and Vue?' />
+
+# ADR 0007: Why do we transpile Web Components for React and Vue?
+
+🗓️ ~2023-04~ 2024-05 · ✍️ [@felipefialho](https://twitter.com/felipefialho_)
+
+## Context
+
+We need to use Web Components in our projects built with Vue and React (Next.js) while maintaining a good Developer Experience (DX).
+
+## Problems
+
+### React
+
+- We need to use `Event Listeners` to handle events and methods when using Web Components
+- React [use `SyntheticEvent` to handle events](https://legacy.reactjs.org/docs/events.html), and we need to convert it to `CustomEvent` to use Web Components
+- Props that use `array` or `object` need to be converted to JSON for compatibility
+- We need to use `ref` to access the Web Component API in React
+- Sometimes, we need to create wrappers to effectively use Web Components in React
+- Overall, the Developer Experience in React is not optimal when working with Web Components
+
+### Vue
+
+- Props that use `array` or `object` need to be converted to JSON for compatibility
+- We need to use `ref` to access the Web Component API in Vue
+- We need to create wrappers to effectively use Web Components in Vue and it is not always straightforward
+
+## Decision
+
+~To address these challenges, we have decided to use Stencil to create a transpiled version of Web Components specifically for React, while keeping the original version for Vue.~
+
+~Currently, we are using Vue 2 (as of April 2022), and the transpiler of Stencil is not compatible with Vue 2, however, Vue provides a good DX and seamless compatibility with Web Components.~
+
+~In the future, when we migrate to Vue 3, we may reevaluate this decision and consider transpiling Web Components for Vue as well.~
```

Comment:
Suggestion: guess we can remove it (the last decision) and add Vue 3 as a requirement in the main README, considering each Atomiun version should have the current decision. 👀

Reply:
wow that looks nice! I didn't notice. 🚀

Reply:
This is better to keep the decision log 🙏